include/deal.II/dofs/dof_handler.h

```diff
@@ -1224,6 +1243,33 @@ DoFHandler<dim,spacedim>::get_fe () const
 
 template <int dim, int spacedim>
 inline
+const FiniteElement<dim,spacedim> &
+DoFHandler<dim,spacedim>::get_finite_element
+(const unsigned int index) const
+{
+  (void) index;
+  Assert(index == 0, ExcMessage("There is only one FiniteElement stored. The index must be zero!"));
+  Assert(selected_fe!=nullptr,
+         ExcMessage("You are trying to access the DoFHandler's FiniteElement object before it has been initialized."));
+  return *selected_fe;
+}
+
+
+
+template <int dim, int spacedim>
+inline
+hp::FECollection<dim,spacedim>
+DoFHandler<dim,spacedim>::get_fe_collection () const
+{
+  Assert(selected_fe!=nullptr,
+         ExcMessage("You are trying to access the DoFHandler's FiniteElement object before it has been initialized."));
+  return hp::FECollection<dim,spacedim>(*selected_fe);
```

Comment:
That's not wrong, but I bet that there are places in the library where we store the result of the corresponding function of the `hp::DoFHandler` class as a reference. The call is also really expensive because `hp::FECollection` *clones* the finite element, and that is an expensive operation for many elements.

So I think the better approach would indeed be to store a clone of the element in an `hp::FECollection`, even for the regular `::DoFHandler`.